kernel/devices/LApic.cpp

```diff
@@ -35,10 +36,28 @@ namespace Kernel::Devices
     }
 
     void LApic::WriteReg(LocalApicRegister reg, uint32_t value) const
-    { sl::MemWrite<uint32_t>(baseAddress + (uint64_t)reg, value); }
+    {
+        if(!x2ModeEnabled)
+        {
+            sl::MemWrite<uint32_t>(baseAddress + (uint64_t)reg, value);
+        }
+        else
+        {
+            CPU::WriteMsr(ExtractX2Offset(reg), value);
+        }
+    }
```

Comment:
Could reverse the order of the if statement? Just a cleanliness thing, I think it reads better `if x do y` rather than `if not x do y`.